### PR TITLE
add pack version to yml file

### PR DESCRIPTION
**Pull request by dansterenson**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3818

## Description
Added/removed the pack version to/from the script as part of the Unify/Split mechanism.
This was done so we can notify the reviewer (at first) what was the original pack the contributor was working on as part of the contribution management flow.

### CODE REVIEW

File: demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py [hunk 1]
```diff
@@ -415,6 +441,23 @@ def test_insert_module_code(mocker, import_to_module):
...
+    When:
+     - calling insert_pack_version
+
+    Then:
+     - Ensure the code returned contains the pack version in it
+    """
+    expected_result = DUMMY_SCRIPT_WITH_PACK_VERSION
+    code = IntegrationScriptUnifier.insert_pack_version(DUMMY_SCRIPT, "1.0.3")
+
+    assert code == expected_result
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/tests/yml_unifier_test.py:458]:
```suggestion
    version_str = "1.0.3"
    assert version_str not in DUMMY_SCRIPT
    assert version_str in IntegrationScriptUnifier.insert_pack_version(DUMMY_SCRIPT, version_str)
```

File: demisto_sdk/commands/split/tests/ymlsplitter_test.py [hunk 2]
```diff
@@ -266,6 +266,7 @@ def test_extract_code(tmpdir):
         assert "from CommonServerPython import *  #" in file_data
         assert file_data[-1] == "\n"
         assert "register_module_line" not in file_data
+        assert "### pack version 1.0.3" not in file_data
```
Comment (dorschw) [inline demisto_sdk/commands/split/tests/ymlsplitter_test.py:269]: add an assertion that "pack version 1.0.3" is in the original file, to make sure this nobody deletes it by mistake

File: demisto_sdk/commands/split/tests/ymlsplitter_test.py [hunk 3]
```diff
@@ -276,6 +277,7 @@ def test_extract_code(tmpdir):
         assert "import demistomock as demisto  #" not in file_data
         assert "from CommonServerPython import *  #" not in file_data
         assert "register_module_line" not in file_data
+        assert "### pack version 1.0.3" not in file_data
```
Comment (dorschw) [inline demisto_sdk/commands/split/tests/ymlsplitter_test.py:280]: add an assertion that "pack version 1.0.3" is in the original file, to make sure this nobody deletes it by mistake

File: demisto_sdk/commands/prepare_content/integration_script_unifier.py [hunk 4]
```diff
@@ -286,6 +287,12 @@ def insert_script_to_yml(
         script_code = IntegrationScriptUnifier.insert_module_code(
             script_code, imports_to_names
         )
+        if pack_version := get_pack_metadata(file_path=str(package_path)).get(
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/integration_script_unifier.py:290]: @ilaner is this the best place to put this functionality, or maybe under `prepare_...`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,7 @@
 * Added a newline at the end of README files generated in **generate-docs**.
 * Added the value `3` (out of bounds) to the `onChangeRepAlg` and `reputationCalc` fields under the `IncidentType` and `GenericType` schemas. **validate** will allow using it now.
 * Added support for running **lint** in multiple native-docker images.
+* Added/removed the pack version to/from the script as part of the Unify/Split mechanism (as part of the contribution management process).
```
Comment (dorschw) [inline CHANGELOG.md:7]:
```suggestion
* Added the pack version to the code files when calling **unify**. The same value is removed when calling **split**.
```

File: demisto_sdk/commands/prepare_content/integration_script_unifier.py [hunk 6]
```diff
@@ -395,6 +402,18 @@ def insert_module_code(script_code: str, import_to_name: Dict[str, str]) -> str:
             script_code = script_code.replace(module_import, module_code)
         return script_code
 
+    @staticmethod
+    def insert_pack_version(script_code: str, pack_version: str) -> str:
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/integration_script_unifier.py:406]: what happens if it's run on code which already has this value?

**Review by dorschw — COMMENTED** [review]
Nice

File: demisto_sdk/commands/prepare_content/integration_script_unifier.py (hunk 6 above)
Comment (dansterenson) [reply demisto_sdk/commands/prepare_content/integration_script_unifier.py:406]: Good point, fixed

File: demisto_sdk/commands/prepare_content/integration_script_unifier.py [hunk 7]
```diff
@@ -395,6 +402,19 @@ def insert_module_code(script_code: str, import_to_name: Dict[str, str]) -> str:
...
+        """
+        Inserts the pack version to the script so it will be easy to know what was the contribution original pack version.
+        :param script_code: The integration code
+        :param pack_version: The pack version
+        :return: The integration script with the pack version appended if needed, otherwise returns the original script
+        """
+        if "### pack version:" in script_code:
+            return script_code
+        pack_version_str = f"\n### pack version: {pack_version}\n"
+        return pack_version_str + script_code
```
Comment (dorschw) [inline demisto_sdk/commands/prepare_content/integration_script_unifier.py:416]:
```suggestion
        return f"\n### pack version: {pack_version}\n{script_code}"
```

**Review by dorschw — APPROVED** [review]

**Comment by dansterenson** [comment]
Extended version of the PR here: https://github.com/demisto/demisto-sdk/pull/2808

**Pull request closed without merging**